mods/gen2/scripts.js

```diff
@@ -117,6 +117,76 @@ exports.BattleScripts = {
 		this.singleEvent('AfterMove', move, null, pokemon, target, move);
 		if (!move.selfSwitch && target.hp > 0) this.runEvent('AfterMoveSelf', pokemon, target, move);
 	},
+	tryMoveHit: function (target, pokemon, move, spreadHit) {
+		let positiveBoostTable = [1, 1.33, 1.66, 2, 2.33, 2.66, 3];
+		let negativeBoostTable = [1, 0.75, 0.6, 0.5, 0.43, 0.36, 0.33];
+		let doSelfDestruct = true;
+		let damage = 0;
+
+		// First, check if the Pokémon is immune to this move.
+		if (move.ignoreImmunity !== true && !move.ignoreImmunity[move.type] && !target.runImmunity(move.type, true)) {
+			return false;
+		}
+
+		if (move.selfdestruct) {
```

Comment:
This is unnecessary because it's handled in `useMove`, but if it weren't, moving it here out of the above block wouldn't cause Explosion users to faint against Ghost types.